tracer/test/Datadog.Trace.Security.IntegrationTests/IAST/IastVerifyScrubberExtensions.cs

```diff
@@ -18,21 +19,51 @@ public static class IastVerifyScrubberExtensions
         private static readonly Regex RequestTaintedRegex = new(@"_dd.iast.telemetry.request.tainted:(\s)*([1-9])(\d*).?(\d*),");
         private static readonly Regex TelemetryExecutedSinks = new(@"_dd\.iast\.telemetry\.executed\.sink\.weak_.+: .{3},");
 
+        private static readonly (Regex RegexPattern, string Replacement) SpanIds = (new Regex(@"(TraceId:|SpanId:|ParentId:)[^,]+,"), string.Empty);
+
+        /*
+             TraceId: Id_1,
+            SpanId: Id_3,
+            Name: aspnet_core_mvc.request,
+            Resource: GET /iast/redirected,
+            Service: Samples.Security.AspNetCore5,
+            Type: web,
+            ParentId: Id_2,
+
+         */
+
         public static VerifySettings AddIastScrubbing(this VerifySettings settings, bool scrubHash = true)
+        {
+            var scrubbers = new List<(Regex RegexPattern, string Replacement)>();
+            if (scrubHash) { scrubbers.Add((HashRegex, string.Empty)); }
+            return AddIastScrubbing(settings, scrubbers);
+        }
+
+        public static VerifySettings AddIastScrubbing(this VerifySettings settings, IEnumerable<(Regex RegexPattern, string Replacement)> extraScrubbers)
         {
             settings.AddRegexScrubber(LocationMsgRegex, string.Empty);
             settings.AddRegexScrubber(ClientIp, string.Empty);
             settings.AddRegexScrubber(NetworkClientIp, string.Empty);
             settings.AddRegexScrubber(RequestTaintedRegex, "_dd.iast.telemetry.request.tainted:,");
             settings.AddRegexScrubber(TelemetryExecutedSinks, string.Empty);
 
-            if (scrubHash)
+            if (extraScrubbers != null)
             {
-                settings.AddRegexScrubber(HashRegex, string.Empty);
+                foreach (var scrubber in extraScrubbers)
+                {
+                    settings.AddRegexScrubber(scrubber.RegexPattern, scrubber.Replacement);
+                }
             }
 
             settings.ScrubEmptyLines();
             return settings;
         }
+
+        public static VerifySettings AddSpanIdsScrubber(this VerifySettings settings)
```

Comment:
I'm surprised this necessary, the ids of the form `Id_1` etc. should be stable between test runs.

Reply:
Not in the case of a redirect. 1 out of 5 runs the redirected span was missing. The other times was the first, hence this fix